### PR TITLE
Facet Test

### PR DESCRIPTION
**Pull request by mcgallan**
To migrate the [Models-TypeSpec](https://github.com/Azure/autorest.csharp/blob/68e44c384b4809bb4c1e989571ba19f081c3ff48/test/TestProjects/Models-TypeSpec/Models-TypeSpec.tsp#L533) from the autorest test project to cadl ranch, and specifically to handle the migration of the facet operation.

### CODE REVIEW

**Review by timotheeguerin — COMMENTED** [review]
what feasture of typespec is this testing

**Comment by mcgallan** [comment]
> what feasture of typespec is this testing

This PR is to add tests for the facet operation.

**Review by weidongxu-microsoft — CHANGES_REQUESTED** [review]
What is facet?

Why it need to be under azure/core? What Azure-specific feature it provides? Non-Azure cannot use this?

File: packages/cadl-ranch-specs/http/azure/core/facet/mockapi.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  value: 15,
+  field: "price",
+};
+
+const responseBody = {
+  field_name: "price",
+  results: [priceResult],
+};
+
+Scenarios.Azure_Core_Facet_getInt32Facets = passOnSuccess(
```
Comment (iscai-msft) [inline packages/cadl-ranch-specs/http/azure/core/facet/mockapi.ts:18]: it's unclear what testing a facet is about testing. What are you trying to test in `azure.core`?

Comment (mcgallan) [reply packages/cadl-ranch-specs/http/azure/core/facet/mockapi.ts:18]: Regarding the facet related operations in Models-TypeSpec, I had some misunderstandings. There are actually no new features. I will close this PR

**Pull request closed without merging**